docs/tutorial.md

```diff
@@ -0,0 +1,231 @@
+          resource("data/airplane"),
+          Csv,
+          SaveMode.ErrorIfExists,
+          options = Map("header" -> "true", "inferSchema" -> "true"))
+          
+        refs += DataUID("raw.weather", "daily") -> new FileSystemDataLink(
+            resource("data/weather/daily"),
+            Csv,
+            SaveMode.ErrorIfExists,
+            options = Map("header" -> "true", "inferSchema" -> "true"))
+        
+        refs += DataUID("raw.weather", "station") -> new FileSystemDataLink(
+            resource("data/weather/station"),
+            Csv,
+            SaveMode.ErrorIfExists,
+            options = Map(
+              "header" -> "true", 
+              "inferSchema" -> "true", 
+              "delimiter"   -> "|"))
+     }
+     
+     private def resource(path: String): String = File.resource(path).pathAsString
+ }
+``` 
+
+Great, now how can you use this? Well, from this moment onwards, you can always access the data sources by just asking the data lake:
+
+```scala
+    val df = AirplaneDatalake("raw" -> "airplane").read()
+```
+This allows you to define all the data sources in your data lake in one single location, and you can use them wherever you need them. You don't need to worry about which format it's in, the exact delimiter to set, ... That's all abstracted away from you. And while this does not add a lot of value if it's just you and a small team, doing a very specific analysis, this can be useful in medium to large organisations, where you have multiple teams of engineers working on the data lake, there is high reuse of data and data scientists just want quick access to all available data. 
+ 
+### Build a data pipeline
+Having a lot of data in your data lake is only valuable once you start doing things with it. This usually involves building some sort of pipeline. There's nothing wrong with simply calling a bunch of Spark transformations in a `run()` function, and simply executing it. And Lighthouse allows you to do exactly that. 
+
+However, in larger projects, this can quickly result in very long, hard to understand, hard to test, and hard to reuse pipelines. So how can we improve on this? By using the `Pipeline` library available in Lighthouse. Let's look at an example:
+
+```scala
+val airlines: SparkFunction[DataFrame] =
+    Sources
+      .fromDataLink(AirplaneDatalake("raw" -> "airplane"))
+      .flatMap(cleanAirlines)
+      .write(SingleFileSink(AirplaneDatalake("clean" -> "airplane")))
+```
+This is a very simple pipeline, that reads the raw `airplane` data, that we defined above, cleans it in the `cleanAirlines` function, and then writes it back to the location `clean -> airplaine` in the data lake. But wait, we haven't defined the clean airplanes yet. Let's do that first. Since we can control how this data source looks like, we can just choose the defaults, which is using the `ORC` format. You can add the following line in your data lake definition:
+
+```scala
```

Comment:
Personally note: I find it confusing that you define your DataUID with 2 strings, but use it as a Map everywhere else.
Please ignore if you don't agree